Core/Base/DataBase.php

```diff
@@ -346,24 +357,24 @@ public function exec($sql)
             }
         }
 
-        return (bool) $result;
+        return $result;
```

Comment:
¿Por qué quitar (bool)?

Reply:
Porqué el tipo de dato que llega en result, ahora ya es de tipo bool, y antes no lo sería.

PHP-CS indica añadir el cast cuando el tipo no es el mismo con el indicado en el return, y te hace sacarlo por innecesario, cuando este ya es correcto.

Todas las rallitas que se vé en el scroll tanto en NetBeans como en PhpStorm, si tienes instalado el PHP-CS, te incluye adicionalmente este tipo de avisos. Conjuntamente con muchos otros, como que no sabe que tipo es una variable (por ejemplo array vs DataBaseWhere[], que es un array igualmente, pero de un tipo concreto).

Esos avisos son los que he ido corrigiendo a medida que iba haciendo la tarea de ir traduciendo, por ese motivo se han juntado más cambios de la cuenta en este PR. Además cuando hago un PR, me muestra errores y warnings, y errores consideraba 4 que realmente no eran tales, y de warnings he reducido como unos 400.

Reply:
Me aparece este cambio como requisito porqué se ha comentado sobre él, pero no se como marcarlo por lo menos para revisar.

This type casting is not necessary, as the argument is of needed type.
The inspection analyzes parameters and return type hints and reports unnecessary type castings.

Los métodos connected, inTransaction y commit, devuelven un booleano y se rellenan en $result, posiblemente antes no se indicaba en el return el tipo concreto, o simplemente había casos en los que no se respondía este tipo, pero ya no es así.